lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1445,6 +1445,127 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                       binder.op, resultType, operand, constAlpha);
                   return success();
                 });
+  patterns.onOp(
+      "LRN", 13, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value operand;
+        int64_t size;
+        float alpha;
+        float beta;
+        float bias;
+        if (binder.tensorOperand(operand) ||
+            binder.tensorResultType(resultType) ||
+            binder.s64IntegerAttr(size, "size", 2) ||
+            binder.f32FloatAttr(alpha, "alpha", 0.0001f) ||
+            binder.f32FloatAttr(beta, "beta", 0.75f) ||
+            binder.f32FloatAttr(bias, "bias", 1.0f))
+          return failure();
+        Type dtype = resultType.getOptionalDtype();
+        Value constAlpha = rewriter.create<Torch::ConstantFloatOp>(
+            binder.getLoc(), rewriter.getType<Torch::FloatType>(),
+            rewriter.getF64FloatAttr(alpha));
+        Value constBeta = rewriter.create<Torch::ConstantFloatOp>(
+            binder.getLoc(), rewriter.getType<Torch::FloatType>(),
+            rewriter.getF64FloatAttr(beta));
+        Value constBias = rewriter.create<Torch::ConstantFloatOp>(
+            binder.getLoc(), rewriter.getType<Torch::FloatType>(),
+            rewriter.getF64FloatAttr(bias));
+        // Please refer to the operator description
+        // for more info on the lowering
+        // https://github.com/onnx/onnx/blob/main/docs/Operators.md#LRN
+
+        // squared = operand^2
+        Location loc = binder.getLoc();
+        Torch::ValueTensorType inTy =
+            cast<Torch::ValueTensorType>(operand.getType());
+        Value sqOperand = rewriter.create<Torch::AtenMulTensorOp>(
+            loc, inTy, operand, operand);
+        // view it as n x 1 x c x d0 x d..
+        if (!inTy.hasSizes()) {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Expected input to have sizes");
+        }
+        ArrayRef<int64_t> inTyShape = inTy.getSizes();
+        if (inTyShape.size() < 3) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "Unsupported: the input dimensions should be >= 3");
+        }
+        SmallVector<int64_t, 5> viewShapeInt{inTyShape[0], 1, inTyShape[1],
+                                             inTyShape[2]};
+        if (inTyShape.size() > 3) {
+          int64_t tailFlatDimSize = 1;
+          for (int64_t dimSize : inTyShape.slice(3)) {
+            tailFlatDimSize *= dimSize;
```

Comment:
Before doing this, you need to make sure whether the shape values are statically known or not, otherwise this would result in issue because of unknown dim size.

Reply:
Good point. I dont think all the dims needed to be statically known (only the one being padded).
I changed the operation view to take -1 instead. It d do the right thing w/o requiring all dims to be statically known unless Im missing something here.